driver-mapping/src/main/java/com/datastax/driver/mapping/Mapper.java

```diff
@@ -357,15 +489,201 @@ public T get(Object... primaryKey) {
+            }
+
+            /**
+             * Get the TTL value configured in the object.
+             *
+             * @return the TTL value.
+             */
+            public int getValue() {
+                return this.ttlValue;
+            }
+
+            void appendTo(Insert.Options usings) {
+                usings.and(QueryBuilder.ttl(QueryBuilder.bindMarker()));
+            }
+
+            void appendTo(Delete.Options usings) {
+                usings.and(QueryBuilder.ttl(QueryBuilder.bindMarker()));
+            }
+
+            void bindInStatement(BoundStatement bs, int i) {
+                bs.setInt(i, this.ttlValue);
+            }
+
+            boolean isValidFor(QueryType qt){
+                return qt == QueryType.SAVE;
+            }
+
+        }
+
+        static class Timestamp extends Option {
+
+            private long tsValue;
+
+            Timestamp(long value) {
+                this.tsValue = value;
+            }
+
+            /**
+             * Get the TIMESTAMP value configured in the object.
+             *
+             * @return the TIMESTAMP value.
+             */
+            public long getValue() {
+                return this.tsValue;
+            }
+
+            void appendTo(Insert.Options usings){
+                usings.and(QueryBuilder.timestamp(QueryBuilder.bindMarker()));
+            }
+
+            void appendTo(Delete.Options usings) {
+                usings.and(QueryBuilder.timestamp(QueryBuilder.bindMarker()));
+            }
+
+
+            boolean isValidFor(QueryType qt){
+                return qt == QueryType.SAVE || qt == QueryType.DEL;
+            }
+
+            void bindInStatement(BoundStatement bs, int i){
+                bs.setLong(i, this.tsValue);
+            }
+
+        }
+
+        /**
+         * Creates a new SaveOptions object for adding a TTL value in a save
+         * or delete operation.
+         *
+         * @param value the value to use for the operation.
+         * @return the SaveOptions object configured to set a TTL value to a
+         * save or delete operation.
+         */
+        public static Option ttl(int value) {
+            return new Ttl(value);
+        }
+
+        /**
+         * Creates a new SaveOptions object for adding a TIMESTAMP value in a save
+         * or delete operation.
+         *
+         * @param value the value to use for the operation.
+         * @return the SaveOptions object configured to set a TIMESTAMP value to a
+         * save or delete operation.
+         */
+        public static Option timestamp(long value) {
+            return new Timestamp(value);
+        }
+
+        abstract void appendTo(Insert.Options usings);
+
+        abstract void appendTo(Delete.Options usings);
+
+        abstract void bindInStatement(BoundStatement bs, int i);
+
+        abstract boolean isValidFor(QueryType qt);
+
+        public int hashCode() {
+            return this.getClass().hashCode();
```

Comment:
It looks like in the context of how Options are used, this hashCode is unique enough as this is only used for the preparedQueries map so the Options objects originally used to create the PreparedStatement are not necessarily the same as the ones used when binding.   So it doesn't appear that this is a problem.

I initially thought this violated the definition of [hashcode](http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#hashCode()), but the doc explicitly states:

> It is not required that if two objects are unequal according to the equals(java.lang.Object) method, then calling the hashCode method on each of the two objects must produce distinct integer results. However, the programmer should be aware that producing distinct integer results for unequal objects may improve the performance of hash tables.

So I think this is ok.